docs/architecture/adr-033-protobuf-inter-module-comm.md

```diff
@@ -83,45 +83,64 @@ and `MsgClient` interfaces and propose this mechanism as a replacement for the e
 this ADR does not necessitate the creation of new protobuf definitions or services. Rather, it leverages the same proto
 based service interfaces already used by clients for inter-module communication.
 
-Using this `QueryClient`/`MsgClient` approach has the following key benefits over keepers:
+Using this `QueryClient`/`MsgClient` approach has the following key benefits over exposing keepers to external modules:
 1. Protobuf types are checked for breaking changes using [buf](https://buf.build/docs/breaking-overview) and because of
 the way protobuf is designed this will give us strong backwards compatibility guarantees while allowing for forward
 evolution.
 2. The separation between the client and server interfaces will allow us to insert permission checking code in between
 the two which checks if one module is authorized to send the specified `Msg` to the other module providing a proper
-object capability system.
+object capability system (see below).
 3. The router for inter-module communication gives us a convenient place to handle rollback of transactions,
 enabling atomicy of operations ([currently a problem](https://github.com/cosmos/cosmos-sdk/issues/8030)). Any failure within a module-to-module call would result in a failure of the entire
 transaction
 
 This mechanism has the added benefits of:
 - reducing boilerplate through code generation, and
-- allowing for modules in other languages either via a VM like CosmWasm or sub-processes using gRPC 
+- allowing for modules in other languages either via a VM like CosmWasm or sub-processes using gRPC
 
 ### Inter-module Communication
 
-To use the `Client` generated by the protobuf compiler we need a `grpc.ClientConn` implementation. For this we introduce
+To use the `Client` generated by the protobuf compiler we need a `grpc.ClientConn` [interface](https://github.com/regen-network/protobuf/blob/cosmos/grpc/types.go#L12)
+implementation. For this we introduce
 a new type, `ModuleKey`, which implements the `grpc.ClientConn` interface. `ModuleKey` can be thought of as the "private
-key" corresponding to a module account, where authentication is provided through use of a special `Invoker()` function, 
+key" corresponding to a module account, where authentication is provided through use of a special `Invoker()` function (gRPC middleware),
```

Comment:
I don't think it's really quite "middleware"

Reply:
right, it's not the best framing. I will remove that.